test/specs/commonTests.js

```diff
@@ -428,13 +431,13 @@ export const implementsIconProp = (Component, requiredProps = {}) => {
         .should.match('i')
     })
 
-    it('accepts an Icon instance', () => {
-      const icon = <Icon className={iconClass} />
-      assertValid(shallow(<Component icon={icon} />))
+    describe('accepts an Icon instance', () => {
+      const icon = <Icon name={iconClass} />
+      assertValid(mount(<Component icon={icon} />))
     })
 
-    it('accepts an icon className string', () => {
-      assertValid(shallow(<Component icon={iconClass} />))
+    describe('accepts an icon className string', () => {
+      assertValid(mount(<Component icon={iconClass} />))
```

Comment:
See above regarding `mount`, I think after that update this can be `shallow` again.